utils/connects.stanza

```diff
@@ -12,41 +12,90 @@ defpackage ocdb/connects :
   import ocdb/bundles
   import ocdb/symbols
   import ocdb/box-symbol
+  import lang-utils
 
-public defn connect-isolate (a:JITXObject, b:JITXObject) :
-  inside pcb-module :
-    eval-when true :
-      match(port-type(a), port-type(b)) :
-        (pa:Bundle, pb:Bundle) :
-          if pa == i2c and pb == i2c :
-            inst buf : ocdb/texas-instruments/ISO1540/component
-            net (a.sda, buf.SDA1)
-            net (a.scl, buf.SCL1)
-            net (b.sda, buf.SDA2)
-            net (b.scl, buf.SCL2)
-            property(buf.VCC1.requires-power) = true
-            property(buf.VCC1.power-request) = [3.3 0.3 0.05]
+; Helper function to check if the ports have bundles and their assigned
+; bundle matches what we expect
+defn ensure-bundles!<?T> (
+  body: () -> ?T, 
+  ports:Seqable<JITXObject>, 
+  bundles:Seqable<Bundle>,
+  action) -> False|T:
+  ; collect the mismatched bundles from ports
+  val errors = 
+    for (p in ports, b in bundles) seq?:
+      match(port-type(p)):
+        (b?:Bundle):
+          if b? != b:
+            One([p, b, b?])  ; case 1: ports exist but do not match
+          else:
+            None()           ; case 2: ports exist and match, we're good to go
+        (_:?):
+          One([p, b, false]) ; case 3: port is not a bundle
+  
+  if empty?(errors):
+    body()
+  else:
+    val buf = StringBuffer()
+    println(buf, "Could not %_. Bundles did not match." % [action])
+    for [p, b, b?] in errors do:
+      println(IndentedStream(buf), 
+        "Expected %_ for port %_. Found %_." % [
+          name(b), ref(p), 
+          name(b? as Bundle) when b? is-not False else port-type(p)
+        ]
+      )
+    fatal(to-string(buf))
 
+; Connect two i2c lines with a buffer. 
+public defn connect-i2c-isolation (
```

Comment:
Please use idiomatic indenting to be consistent with the rest of the code base:
```
 public defn connect-i2c-isolation (
   x:JITXObject,
   y:JITXObject,
   buffer:Instantiable
 ):
   within ensure-bundles!([x, y], [i2c, i2c], "call connect-isolation"):
     inside pcb-module:
       public inst buf: buffer
       net (x, buf.p[1])
       net (y, buf.p[2])
       buf
```
should be this:
```
 public defn connect-i2c-isolation (x:JITXObject,
                                    y:JITXObject,
                                    buffer:Instantiable):
   within ensure-bundles!([x, y], [i2c, i2c], "call connect-isolation"):
     inside pcb-module:
       public inst buf: buffer
       net (x, buf.p[1])
       net (y, buf.p[2])
       buf
```